### PR TITLE
Refactor dev environment and toolchain

### LINKED ISSUE #100: Migrate to dep
Glide has officially gone into maintenance mode and is [recommending](https://github.com/Masterminds/glide#golang-dep) users migrate to [dep](https://github.com/golang/dep) for managing dependencies.

### PR DESCRIPTION
**Pull request by dragan**
- Migrate from glide to dep
  - Include vendor directory so we are go get/install able
  - Use make for all build steps
  - Merge building and creating docker images into one file instead
    of dispersed between many files.
  - Decrease security surface area by creating image FROM scratch.
  - Even though we can build multi-platform, let's not distribute
    quite yet.
  - Remove docker-compose, dev, vault, and etc. as those will be
    provided by a different repo/process.
  - Modify CI configuration to use new build process.
  - Docker images are now pushed to scipian org in quay.

Closes #100

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,55 +1,275 @@
...
+	    -v "$$(pwd)/.go:/go"                                                      \
+	    -v "$$(pwd):/go/src/$(PKG)"                                               \
+	    -v "$$(pwd)/bin/$(OS)-$(ARCH):/go/bin"                                    \
+	    -v "$$(pwd)/bin/$(OS)-$(ARCH):/go/bin/$(OS)_$(ARCH)"                      \
+	    -v "$$(pwd)/.go/std/$(OS)-$(ARCH):/usr/local/go/pkg/$(OS)_$(ARCH)_static" \
+	    -w /go/src/$(PKG)                                                         \
+	    $(BUILD_IMAGE)                                                            \
+	    /bin/sh -c "                                                              \
+	        ARCH=$(ARCH)                                                          \
+                OS=$(OS)                                                              \
```
Comment (nicklathe) [inline Makefile]: Super nit: The `\` is one space further to the right than all the others.

Comment (dragan) [reply Makefile]: I think that's GitHub, because the file doesn't show the extra space.

Comment (dragan) [reply Makefile]: Found the issue.

**Comment by dragan** [comment]
I did some testing and I missed installing git into the image. Once I get it installed, this should be good to go. I'll submit a PR to scipian/infra for our local dev environment too.

**Pull request merged**